hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/service/HxInsightClient.java

```diff
@@ -71,14 +75,15 @@ public String askQuestion(Question question)
 
             HttpResponse<String> httpResponse = client.send(request, BodyHandlers.ofString());
 
-            throwExceptionOnUnexpectedStatusCode(httpResponse.statusCode(), EXPECTED_STATUS_CODE);
+            ensureThat(httpResponse.statusCode() == EXPECTED_STATUS_CODE,
+                    () -> new WebScriptException(httpResponse.statusCode(), "Request to hxi failed"));
 
             return objectMapper.readValue(httpResponse.body(), QuestionResponse.class)
                     .questionId();
         }
         catch (IOException | InterruptedException e)
         {
-            throw new HxInsightConnectorRuntimeException("Failed to ask question", e);
+            throw new WebScriptException(INTERNAL_SERVER_ERROR, "Failed to ask question", e);
```

Comment:
I think we want something that's more obviously retriable - like 502 Service Unavailable or 503 Bad Gateway.

Reply:
Since we only have one client for this private API then maybe we should just ask @MichalKinas what status codes they retry on.

Reply:
503 Service Unavailable would fit here best IMO.

Reply:
I asked @AleksanderSklorz and he told me that we should just pass status code to them, so I'll leave it as it is

Reply:
I have one question for clarification - which status code will be used to let us know we should still retry call and which that we should stop retrying because there is something wrong on backend? Just to avoid situation that we will be doing infinite number of calls

Reply:
I think you should change this one because it's returning a 500 for a network issue. I don't think ADW will retry on a 500, but hopefully they do with a 503.

Reply:
We may also want to think about a generalized error message so that it can be i18-ed.